python/lsst/ap/association/transformDiaSourceCatalog.py

```diff
@@ -182,8 +183,11 @@ def run(self,
                             ParquetTable(dataFrame=diaSourceDf),
                             self.funcs,
                             dataId=None).df
+        # The Ra/DecColumn functors preserve the coord_ra/dec original columns.
+        # Since we don't need these and keeping them causes a DB insert crash
+        # we drop them from the DataFrame before returning out value.
         return pipeBase.Struct(
-            diaSourceTable=df
+            diaSourceTable=df.drop(columns=["coord_ra", "coord_dec"]),
```

Comment:
Checking the docs, `df.drop()` defaults to returning the modified object. That results in a new copy of the data frame. I think what you want here is to make the `df.drop()` call before `return`, and pass `inplace=True` so that it modifies the dataframe in place without making a copy. See:

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.drop.html

Reply:
I understand that's what happening. What is your concern here of returning the copy of the dataframe that is computed in this method?

Reply:
A general avoidance of creating unnecessary copies.

Though I've now spent some time on stackoverflow and some pandas docs, and it looks like an extra copy is typically created in this case even with `inplace=True` (there's a proposal to deprecate `inplace`: https://github.com/pandas-dev/pandas/issues/16529 ). This is surprising behavior, coming from the C++ side of the stack where "don't make unnecessary copies" was a big design goal.

So, I guess leave the code here as-is, and now I wonder whether we shouldn't mention this somewhere in the dev guide since we're starting to use pandas in more places?